x-pack/plugins/reporting/server/export_types/csv/create_job.ts

```diff
@@ -18,10 +19,10 @@ export const createJobFnFactory: CreateJobFnFactory<CreateJobFn<
     const serializedEncryptedHeaders = await crypto.encrypt(request.headers);
 
     const savedObjectsClient = context.core.savedObjects.client;
-    const indexPatternSavedObject = await savedObjectsClient.get(
+    const indexPatternSavedObject = ((await savedObjectsClient.get(
       'index-pattern',
       jobParams.indexPatternId
-    );
+    )) as unknown) as IndexPatternSavedObject; // FIXME
```

Comment:
Our definition of IndexPatternSavedObject does not line up to the definition created by @elastic/kibana-app-arch. 

This problem happened because we added types for these objects when the App Arch code was not in Typescript. 

The FIXME can be addressed, and all the custom Reporting types can be removed, by rewriting the internals of Export CSV to use SearchSource on the server side.

That would allow Reporting to only keep track of the ID of the saved search in job params - no more reverse engineering how to derive a query from an index pattern object.

Reply:
It's the same situation as these existing lines of code: https://github.com/elastic/kibana/blob/7.9/x-pack/plugins/reporting/server/export_types/csv_from_savedobject/lib/get_csv_job.ts#L129-L130

Reply:
Good to know!